ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -3359,7 +3359,7 @@ public Object post(Object t) {
         ASTNode root = (ASTNode) t;
         if (root.getType() == HiveParser.TOK_FUNCTION) {
           ASTNode func = (ASTNode) ParseDriver.adaptor.getChild(root, 0);
-          if (func.getText().equals("grouping") && func.getChildCount() == 0) {
+          if (func.getText().equalsIgnoreCase("grouping") && func.getChildCount() == 0) {
```

Comment:
Could you please change this to
```
"grouping".equalsIgnoreCase(func.getText())
```
it is null safe.

Reply:
Good point. Thanks!